.github/scripts/release-against-charts.sh

```diff
@@ -27,34 +27,36 @@ if [ ! -f bin/charts-build-scripts ]; then
     make pull-scripts
 fi
 
-find ./packages/fleet/ -type f -exec sed -i -e "s/${PREV_FLEET_VERSION}/${NEW_FLEET_VERSION}/g" {} \;
-find ./packages/fleet/ -type f -exec sed -i -e "s/version: ${PREV_CHART_VERSION}/version: ${NEW_CHART_VERSION}/g" {} \;
+if grep -q "version: ${PREV_CHART_VERSION}" ./packages/fleet/fleet/package.yaml && grep -q "${PREV_FLEET_VERSION}" ./packages/fleet/fleet/package.yaml; then
+    find ./packages/fleet/ -type f -exec sed -i -e "s/${PREV_FLEET_VERSION}/${NEW_FLEET_VERSION}/g" {} \;
+    find ./packages/fleet/ -type f -exec sed -i -e "s/version: ${PREV_CHART_VERSION}/version: ${NEW_CHART_VERSION}/g" {} \;
+else
+    echo "Previous Fleet version references do not exist in ./packages/fleet/ so replacing it with the new version is not possible. Exiting..."
+    exit 1
+fi
 
 if [ "${REPLACE}" == "true" ]; then
-    sed -i -e "s/${PREV_CHART_VERSION}+up${PREV_FLEET_VERSION}/${NEW_CHART_VERSION}+up${NEW_FLEET_VERSION}/g" release.yaml
+    for i in fleet fleet-crd fleet-agent; do
+        yq --inplace "del( .${i}.[] | select(. == \"${PREV_CHART_VERSION}+up${PREV_FLEET_VERSION}\") )" release.yaml
+        yq --inplace ".${i} += \"${NEW_CHART_VERSION}+up${NEW_FLEET_VERSION}\"" release.yaml
+    done
 else
-    if grep -q "fleet:" release.yaml; then
-        sed -i -e "s/- ${PREV_CHART_VERSION}+up${PREV_FLEET_VERSION}/- ${PREV_CHART_VERSION}+up${PREV_FLEET_VERSION}\n- ${NEW_CHART_VERSION}+up${NEW_FLEET_VERSION}/g" release.yaml
-    else
-        cat <<< "fleet:
-- ${PREV_CHART_VERSION}+up${PREV_FLEET_VERSION}
-- ${NEW_CHART_VERSION}+up${NEW_FLEET_VERSION}
-fleet-agent:
-- ${PREV_CHART_VERSION}+up${PREV_FLEET_VERSION}
-- ${NEW_CHART_VERSION}+up${NEW_FLEET_VERSION}
-fleet-crd:
-- ${PREV_CHART_VERSION}+up${PREV_FLEET_VERSION}
-- ${NEW_CHART_VERSION}+up${NEW_FLEET_VERSION}" >> release.yaml
-    # remove empty line above fleet
-    sed -i -z -e  "s/[[:space:]]*\nfleet:/\nfleet:/g" release.yaml
-    fi
+    for i in fleet fleet-crd fleet-agent; do
+        yq --inplace ".${i} += \"${NEW_CHART_VERSION}+up${NEW_FLEET_VERSION}\"" release.yaml
+    done
 fi
 
+# Follow the currently applied no identation style for arrays in release.yaml,
+# which is not possible with yq v4 atm: https://github.com/mikefarah/yq/issues/1309
+sed -i "s/^  -/-/g" release.yaml
```

Comment:
nit: I suppose the `g` may not be necessary in this `sed` expression since it deals with line beginnings.